tests/ui/borrowck/borrowck-vec-pattern-nesting.rs

```diff
@@ -8,6 +8,7 @@ fn a() {
         //~^ NOTE `vec[_]` is borrowed here
             vec[0] = Box::new(4); //~ ERROR cannot assign
             //~^ NOTE `vec[_]` is assigned to here
+            //~| NOTE in this expansion of desugaring of drop and replace
```

Comment:
Should `DesugaringKind::Replace` be filtered out from diagnostic notes?

Reply:
It's not shown in human readable output but only in json, like other desugarings (`for`, `?`, ...). I think it makes sense they all have the same behavior